[WIP] Some refactoring

igor-raits opened the pull request:
- Better validation
- Less useless information in terminal
- Support for building one package just without any settings
- Support specifying git directory for speeding up local building
- Support for building using rpmbuild
- RPM package
- Cleanups, speedups

Commit:
refactoring

* Better validation
* Support for building one package without any YAML definitions
* Support specifying existing git directory
* Now date for changelog is same as date of commit
* Cleanups, speedups

Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>

Commit:
git: fix crash if timestamp requested for tag

Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>

Commit:
git: strip 'version' prefix

For example, in FVWM they tag releases like "version-2_6_6".

Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>

Commit:
builders: add rpmbuild builder

Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>

Commit:
main: require chroot only where it's supported

Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>

Commit:
component: be more strict about regex in make_srpm()

Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>

Commit:
add .travis.yml

Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>

Commit:
spec: add some hard requirements like tar, xz, rpmbuild

Signed-off-by: Igor Gnatenko <ignatenko@redhat.com>

igor-raits committed:
setup.py: explicitly set entry point function

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>

igor-raits committed:
component: move import rpm into function

do import rpm only in functions as it is not distributed by PyPI
so we can't test under Travis.

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>

igor-raits committed:
travis: explicitly disable sudo

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>

igor-raits committed:
main: always configure logger

otherwise we can get incostitent style of logs when --log is set and
when it's not.

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>

igor-raits committed:
main: always use relative imports

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>

igor-raits committed:
don't pollute stdout with verbose output from rpmbuild

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>

igor-raits committed:
trivial: make some lines not that long

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>

igor-raits commented:
@mluscon @MichaelMraka @jsilhan wanna give a try?

jsilhan commented:
> Support for building one package just without any settings
> Support specifying git directory for speeding up local building

Is there any instruction (README, example, docs, ...) how to take advantage of this?

igor-raits commented:
@jsilhan definitely yes. `rpm-gitoverlay --help` ;)

igor-raits committed:
spec: drop unneeded BRs

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>

igor-raits committed:
spec: don't run coverage tests while building

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>

igor-raits committed:
main: show warning if builder is rpmbuild and more than 1 component

Well, if there are dependencies between components we can't handle them
by plain rpmbuild without installing RPMs into system which is dangerous.

Signed-off-by: Igor Gnatenko <i.gnatenko.brain@gmail.com>
